plugins/inputs/smart/smart.go

```diff
@@ -491,12 +935,30 @@ func parseTemperature(fields, deviceFields map[string]interface{}, str string) e
 	return nil
 }
 
+func parseTemperatureSensor(fields, deviceFields map[string]interface{}, str string) error {
+	var temp int64
+	if _, err := fmt.Sscanf(str, "%d C", &temp); err != nil {
+		return err
+	}
+
+	fields["raw_value"] = temp
+
+	return nil
+}
+
 func init() {
+	// Set LC_NUMERIC to uniform numeric output from cli tools
+	_ = os.Setenv("LC_NUMERIC", "en_US.UTF-8")
+
 	inputs.Add("smart", func() telegraf.Input {
 		m := NewSmart()
-		path, _ := exec.LookPath("smartctl")
-		if len(path) > 0 {
-			m.Path = path
+		pathSmartctl, _ := exec.LookPath("smartctl")
```

Comment:
plugins can have an `Init() error` function. looking up the path names (and erroring if they're missing) might be better there.

Reply:
@ssoroka 
It depends. If I return any error in `Init() error`, Telegraf won't start. In approach we proposed (which is aligned with previous version), plugin won't gather anything but Telegraf (with other plugins) will work properly.

It is up to you :)

Reply:
@ssoroka 
It depends. If I return any error in `Init() error`, Telegraf won't start. In approach we proposed (which is aligned with previous version), plugin won't gather anything but Telegraf (with other plugins) will work properly.

It is up to you :)

Reply:
Did as you said! Thanks for guidance!